torchvision/models/_api.py

```diff
@@ -140,3 +164,63 @@ def _get_enum_from_fn(fn: Callable) -> WeightsEnum:
         )
 
     return cast(WeightsEnum, weights_enum)
+
+
+M = TypeVar("M", bound=nn.Module)
+
+BUILTIN_MODELS = {}
+
+
+def register_model(name: Optional[str] = None) -> Callable[[Callable[..., M]], Callable[..., M]]:
+    def wrapper(fn: Callable[..., M]) -> Callable[..., M]:
+        key = name if name is not None else fn.__name__
+        if key in BUILTIN_MODELS:
+            raise ValueError(f"An entry is already registered under the name '{key}'.")
+        BUILTIN_MODELS[key] = fn
+        return fn
+
+    return wrapper
+
+
+def list_models(module: Optional[ModuleType] = None) -> List[str]:
+    """
+    Returns a list with the names of registered models.
+
+    .. betastatus:: function
+
+    Args:
+        module (ModuleType, optional): The module from which we want to extract the available models.
+
+    Returns:
+        models (list): A list with the names of available models.
+    """
+    models = [
+        k for k, v in BUILTIN_MODELS.items() if module is None or v.__module__.rsplit(".", 1)[0] == module.__name__
+    ]
+    return sorted(models)
+
+
+def find_model(name: str) -> Callable[..., M]:
+    name = name.lower()
+    try:
+        fn = BUILTIN_MODELS[name]
+    except KeyError:
+        raise ValueError(f"Unknown model {name}")
+    return fn
+
+
+def get_model(name: str, **config: Any) -> M:
```

Comment:
+1 to Nicolas naming suggestion. My preference would be `build_model` and `get_model_builder`. This can be done in a separate PR and align to datasets or together with any changes that might arise from community feedback. 
Shouldn't have missed it in the RFC.

Reply:
Sounds good. Let's release is as beta to get input and we can promote to stable once we get enough positive feedback from the community. Concerning the details of this comment, the `get_model()` returns an initialized model it doesn't return the model builder. It's the `find_model()` that returns the model builder, but that's an internal method on the `_api` module and not exposed publicly. At any case, I agree it's worth reviewing the names on a follow up and ensure they are aligned with the mechanism proposed on the datasets.

Reply:
I meant `get_model` -> `build_model` and `find_model` -> `get_model_builder`.